pkg/storage/s3.go

```diff
@@ -440,3 +452,64 @@ func (r *s3ObjectReader) Seek(offset int64, whence int) (int64, error) {
 	r.pos = realOffset
 	return realOffset, nil
 }
+
+// ListObject list `maxkeys` objects with specify path.
+func (rs *S3Storage) ListObject(ctx context.Context, name string, maxKeys int64) (*s3.ListObjectsV2Output, error) {
+	input := &s3.ListObjectsV2Input{
+		Bucket:  aws.String(rs.options.Bucket),
+		Prefix:  aws.String(rs.options.Prefix + name),
+		MaxKeys: aws.Int64(maxKeys),
+	}
+	return rs.svc.ListObjectsV2WithContext(ctx, input)
+}
+
+// CreateMultipartUpload create multi upload request.
+func (rs *S3Storage) CreateMultipartUpload(ctx context.Context, name string) (*s3.CreateMultipartUploadOutput, error) {
```

Comment:
how CDC is using these 3 methods (CreateMultipartUpload, UploadPart, CompleteMultipartUpload)? can we combine them into one API and use [Uploader](https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.34.3/service/s3/s3manager?tab=doc#Uploader) instead?

Reply:
CDC use these API int this [PR](https://github.com/pingcap/ticdc/pull/826/files#diff-9ee946c73b42d31cb07ff1fcb4defbaeR348), when flush every batch events, if the size of generate date > 5M. we will use multi-upload to upload data as a chunk. when the upload data size reached 100M, we complete chunks to a file.

Let me see Uploader first.

Reply:
the bottom line is it should not return a `*s3.CreateMultipartUploadOutput` which isn't applicable to GCS or Local.

Reply:
OK, I add a uploader interface, please take a look